src/workflow-engine/charms/temporal-worker/oci_factory/activities/find_images_to_update.py

```diff
@@ -221,10 +221,12 @@ def find_released_revisions(releases_json: dict) -> list:
                             f"Track {to_track} is missing its end-of-" "life field"
                         )
 
+                # filter out the tracks with releases out of EOL
+                release_to = {k : v for k, v in release_to.items() if "end-of-life" in v}
```

Comment:
The code is correct, but I'd rather not loop over the release_to dict again unnecessarily. Instead, you can add an `else` block to the inner `if` block above that check for eol against current date, and delete `to_track` key from `release_to` dict there.